test/snoowrap.spec.js

```diff
@@ -19,6 +19,7 @@ describe('snoowrap', function () {
     oauthInfo = process.env.CI ? {
       user_agent: process.env.SNOOWRAP_USER_AGENT,
       client_id: process.env.SNOOWRAP_CLIENT_ID,
+      client_id_installed: process.env.SNOOWRAP_CLIENT_ID_INSTALLED,
```

Comment:
If this requires a new property to be added to `oauth_info.json` as part of local setup, can you also add it to the list [here](https://github.com/not-an-aardvark/snoowrap/blob/cd2cc85fdc6354e641a458c192795e8784c0c675/src/README.md)?

Reply:
As far as I can tell there isn't another way to test additional grant types (installed & client credentials) without setting up an app for each. Thoughts?

I should probably add another for client credentials as well and update the tests to try both grant types.

Reply:
I guess we should set up additional apps then. Ideally it would be nice to make the setup process easier, but I'm not sure how to avoid it in this case.

Could the regular "personal script" credentials used for other tests also be used for the `client_credentials` grant? If so, it would only be necessary to add one additional set of credentials.

If the code for both application-only oauth flows is exactly the same aside from passing a different user-provided string to the server, I think we could also skip testing one of them and just test the other one.

Reply:
They both take a different second parameter

So I could add two different "app only" setups like below:

```json
  "application_only_auth": {
    "client_credentials": {
      "client_id": "[CLIENT ID]",
      "client_secret": "[CLIENT SECRET]"
    },
    "installed_app": {
      "client_id": "[CLIENT ID]",
      "device_id": [DEVICE_ID]"
    }
  }
```

or combine them but have the tests just use the params it needs

```json
"application_only_auth": {
   "client_id": "[CLIENT ID]",
   "client_secret": "[CLIENT SECRET]",
   "device_id": [DEVICE_ID]"
}
```

Reply:
As far as I know, client IDs/secrets are only categorized by app type on reddit, not by whether they use application-only oauth. Since there is already a "personal use script" token used for regular authenticated requests in the tests, could the same token be used for one of the application-only oauth flows?

In other words, I'm wondering if it would make more sense to label the test tokens as "personal use script token", "installed app token", etc. rather than having a separate token of the same type that only gets used for application-only oauth.

Reply:
That makes sense. Guess I didn't realize they were reusable between the two. Simplified this to only need one extra token for the installed client test.